File: pec-observation-app/ios-app/Sources/PECObservation/Ports/SchoolRepositoryPort.swift
import Foundation

public protocol SchoolRepositoryPort {
    func save(_ school: School) throws
    func fetchAll() throws -> [School]
    func fetchById(_ id: UUID) throws -> School?
    func delete(_ id: UUID) throws
}

public protocol TeacherRepositoryPort {
    func save(_ teacher: Teacher) throws
    func fetchAll() throws -> [Teacher]
    func fetchBySchool(_ schoolId: UUID) throws -> [Teacher]
    func fetchById(_ id: UUID) throws -> Teacher?
    func delete(_ id: UUID) throws
}

File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/ports/SchoolRepositoryPort.kt
package br.gov.educacao.sp.pecobservation.ports

import br.gov.educacao.sp.pecobservation.domain.entities.School
import br.gov.educacao.sp.pecobservation.domain.entities.Teacher
import kotlinx.coroutines.flow.Flow
import java.util.UUID

interface SchoolRepositoryPort {
    fun observeAll(): Flow<List<School>>
    suspend fun save(school: School)
    suspend fun fetchById(id: UUID): School?
}

interface TeacherRepositoryPort {
    fun observeBySchool(schoolId: UUID): Flow<List<Teacher>>
    suspend fun save(teacher: Teacher)
    suspend fun fetchById(id: UUID): Teacher?
}
